protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,94 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"errors"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+var (
+	ErrNoFilterSubjects          = errors.New("no filter subjects were given")
+	ErrMoreThanOneStream         = errors.New("more than one stream for given filter subjects")
+	ErrNoConsumerConfig          = errors.New("no consumer config was given")
+	ErrMoreThanOneConsumerConfig = errors.New("more than one consumer config given")
+)
+
+// NatsOptions is a helper function to group a variadic nats.ProtocolOption into
+// []nats.Option that can be used by either Sender, Consumer or Protocol
+func NatsOptions(opts ...nats.Option) []nats.Option {
+	return opts
+}
+
+// ProtocolOption is the function signature required to be considered an nats.ProtocolOption.
+type ProtocolOption func(*Protocol) error
+
+func WithConsumerOptions(opts ...ConsumerOption) ProtocolOption {
+	return func(p *Protocol) error {
+		p.consumerOptions = opts
+		return nil
+	}
+}
+
+func WithSenderOptions(opts ...SenderOption) ProtocolOption {
+	return func(p *Protocol) error {
+		p.senderOptions = opts
+		return nil
+	}
+}
+
+// WithPublishOptions configures the Sender
+func WithPublishOptions(publishOpts []jetstream.PublishOpt) SenderOption {
+	return func(s *Sender) error {
+		s.PublishOpts = publishOpts
+		return nil
+	}
+}
+
+// WithConsumerConfig configures the Consumer with the given config
+func WithConsumerConfig(consumerConfig *jetstream.ConsumerConfig) ConsumerOption {
```

Comment:
I was able to combine the SenderOptions, CustomerOptions, and JetstreamOptions into the one and only  ProtocolOptions.

I was not able to combine the NatsOptions because the NatsOptions are used early in the process to create the connection.